segmentstore/server/src/main/java/io/pravega/segmentstore/server/DirectSegmentAccess.java

```diff
@@ -51,14 +51,34 @@
      */
     CompletableFuture<Long> append(BufferView data, Collection<AttributeUpdate> attributeUpdates, Duration timeout);
 
+    /**
+     * Appends a range of bytes at the end of the Segment and atomically updates the given attributes. The byte range
+     * will be appended as a contiguous block, however there is no guarantee of ordering between different calls to this
+     * method.
+     * @see io.pravega.segmentstore.contracts.StreamSegmentStore#append(String, BufferView, Collection, Duration)
+     *
+     * @param data             The data to add.
+     * @param attributeUpdates A Collection of Attribute-Values to set or update. May be null (which indicates no updates).
+     *                         See Notes about AttributeUpdates in the interface Javadoc.
+     * @param offset           The expected offset that the append will occur at.
```

Comment:
"The expected length of the segment. If the current length of the Segment does not equal this value, the operation will fail with a {@link BadOffsetException}."